### PR TITLE
Remove libVersions

### PR DESCRIPTION
**Pull request by pkoenig10**
Use `versions.props` to manage all dependencies.

### CODE REVIEW

File: gradle/protobuf.gradle [hunk 1]
```diff
@@ -0,0 +1,13 @@
+apply plugin: 'com.google.protobuf'
+
+protobuf {
+    protoc {
+        artifact = 'com.google.protobuf:protoc:3.21.2'
```
Comment (pkoenig10) [inline gradle/protobuf.gradle:5]: This shouldn't be necessary. GCV can recommend a version here.

This works on `protobuf-gradle-plugin` version 0.8.12, but not on 0.8.13. On 0.8.13 and newer, `generateProto` fails with:
```
Expected configuration ':atlasdb-client-protobufs:protobufToolsLocator_protoc' files to contain exactly one file, however, it contains no files.
```

If you manually inspect the `protobufToolsLocator_protoc` files, it definitely does have a single file. I think there's something funky caused by the fact that the plugin stores this as a file collection. I think the regression happened in https://github.com/google/protobuf-gradle-plugin/pull/408, but I'm not sure.

**Comment by pkoenig10** [comment]
I'll fix up the conflicts once the PR is approved.

**Review by Jolyon-S — COMMENTED** [review]
I've reviewed 46/55 files - these are the ones where there are straight substitutions, and all looks good. I just want to do a second pass where I validate that the changes that aren't straight substitutions aren't contentious.

File: atlasdb-dbkvs/build.gradle [hunk 2]
```diff
@@ -31,6 +31,8 @@ dependencies {
   implementation project(':commons-executors')
   implementation project(':timestamp-api')
 
+  runtimeOnly 'com.oracle.database.jdbc:ojdbc11'
```
Comment (Jolyon-S) [inline atlasdb-dbkvs/build.gradle:34]: What caused this change?

File: gradle/protobuf.gradle (hunk 1 above)
Comment (Jolyon-S) [reply gradle/protobuf.gradle:5]: Won't specifying our protobuf version directly here make upgrading protobuf's version harder (i.e. we'll need to update in multiple placers)?

File: gradle/protobuf.gradle [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+    }
+}
+
+dependencies {
+    api 'com.google.protobuf:protobuf-java'
```
Comment (Jolyon-S) [inline gradle/protobuf.gradle:10]: Does this need to be api? We had this as implementation before.

File: examples/profile-client-protobufs/build.gradle [hunk 4]
```diff
@@ -1,20 +1,4 @@
 apply from: "../../gradle/shared.gradle"
+apply from: "../../gradle/protobuf.gradle"
 
 group = 'com.palantir.atlasdb.examples'
-
-apply plugin: 'com.google.protobuf'
-
-// The protoc version atlasdb uses does not yet support Apple M1 aarch64
-// (see https://github.com/protocolbuffers/protobuf/issues/8062), so use x86_64.
```
Comment (Jolyon-S) [inline examples/profile-client-protobufs/build.gradle:8]: Have you tested this change on an M1?

**Review by Jolyon-S — COMMENTED** [review]
Thanks @pkoenig10! I just want to sanity check a couple of changes before we go, but otherwise looks good.

File: gradle/protobuf.gradle (hunk 1 above)
Comment (pkoenig10) [reply gradle/protobuf.gradle:5]: The version of `protoc` was already specified separately before this PR in `libVersions.protoc`. This PR just moves that from `libVersions` to here.

File: gradle/protobuf.gradle (hunk 3 above)
Comment (pkoenig10) [reply gradle/protobuf.gradle:10]: Yes, `com.google.protobuf` are exposed in multiple places in the generated classes. For example, `StreamMetadataOrBuilder.getHash`.

File: atlasdb-dbkvs/build.gradle (hunk 2 above)
Comment (pkoenig10) [reply atlasdb-dbkvs/build.gradle:34]: I should remove this. We get these runtime dependencies from `atlasdb-dbkvs-hikari`, which is where we actually use these classes at runtime. AFAICT the `atlasdb-dbkvs` package itself doesn't use these classes at runtime.

File: examples/profile-client-protobufs/build.gradle (hunk 4 above)
Comment (pkoenig10) [reply examples/profile-client-protobufs/build.gradle:8]: Yes, running `generateProto` succeeds on an M1.

**Review by Jolyon-S — APPROVED** [review]
Thank you @pkoenig10!

**Pull request merged**